amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisher.java

```diff
@@ -222,8 +224,8 @@ public PrefetchRecordsPublisher(final int maxPendingProcessRecordsInput, final i
         this.defaultGetRecordsCacheDaemon = new DefaultGetRecordsCacheDaemon();
         Validate.notEmpty(operation, "Operation cannot be empty");
         this.operation = operation;
-        this.streamAndShardId =
-                this.getRecordsRetrievalStrategy.getDataFetcher().getStreamIdentifier().serialize() + ":" + shardId;
+        this.streamId = this.getRecordsRetrievalStrategy.getDataFetcher().getStreamIdentifier();
```

Comment:
```suggestion
        this.streamId = this.getRecordsRetrievalStrategy.dataFetcher().getStreamIdentifier();
```

Reply:
This change needs to be done during merge, as the interface method is not available as of now